core/Plugin/Manager.php

```diff
@@ -1334,6 +1336,10 @@ private function sortPluginsSameOrderAsGlobalConfig(array $plugins)
         if (empty($global)) {
             return $plugins;
         }
+
+        // we need to make sure a possibly disabled plugin will be still loaded before any 3rd party plugin
+        $global = array_merge($global, $this->corePluginsDisabledByDefault);
```

Comment:
I had to add this as I created a 3rd party plugin that required the plugin `MobileAppType`. The 3rd party plugin was always enabled before `MobileAppType` so the missing dependency check did not quite work and the 3rd party plugin was always automatically disabled.